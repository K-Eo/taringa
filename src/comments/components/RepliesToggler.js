import React from 'react'

const RepliesToggler = ({ isVisible, totalCount, onToggleVisibility }) => {
  if (totalCount === 0) {
    return null
  }

  return (
    <button
      className="btn btn-link text-dark Commentable-replies-toggler mb-2"
      onClick={onToggleVisibility}
    >
      {isVisible ? (
        <span>
          {`Ocultar ${totalCount.pluralize('respuesta', null, true)} `}
          <i className="fa fa-chevron-up" />
        </span>
      ) : (
        <span>
          {`Ver ${totalCount.pluralize('respuesta', null, totalCount === 1)} `}
          <i className="fa fa-chevron-down" />
        </span>
      )}
    </button>
  )
}

RepliesToggler.defaultProps = {
  isVisible: false,
  onToggleVisibility: () => {},
  totalCount: 0,
}

export default RepliesToggler
