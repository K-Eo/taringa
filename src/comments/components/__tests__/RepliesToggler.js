import React from 'react'

import RepliesToggler from '../RepliesToggler'

describe('Replies Toggler', () => {
  it('exists', () => {
    expect(RepliesToggler).toBeDefined()
  })

  it('renders nothing if totalCount is zero', () => {
    const wrapper = shallow(<RepliesToggler />)

    expect(wrapper).toMatchSnapshot()
  })

  it('changes message on visibility', () => {
    const wrapper = shallow(<RepliesToggler totalCount={1} />)

    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ isVisible: true })
    expect(wrapper).toMatchSnapshot()
  })

  it('handles onToggleVisibility', () => {
    const mock = jest.fn()

    const wrapper = shallow(
      <RepliesToggler totalCount={1} onToggleVisibility={mock} />
    )

    wrapper.find('button').simulate('click')

    expect(mock).toHaveBeenCalled()
  })
})
