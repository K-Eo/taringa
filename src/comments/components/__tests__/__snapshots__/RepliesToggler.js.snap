// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Replies Toggler changes message on visibility 1`] = `
<button
  className="btn btn-link text-dark Commentable-replies-toggler mb-2"
  onClick={[Function]}
>
  <span>
    Ver respuesta 
    <i
      className="fa fa-chevron-down"
    />
  </span>
</button>
`;

exports[`Replies Toggler changes message on visibility 2`] = `
<button
  className="btn btn-link text-dark Commentable-replies-toggler mb-2"
  onClick={[Function]}
>
  <span>
    Ocultar respuesta 
    <i
      className="fa fa-chevron-up"
    />
  </span>
</button>
`;

exports[`Replies Toggler renders nothing if totalCount is zero 1`] = `""`;
